pkg/webhook/patch.go

```diff
@@ -518,6 +523,19 @@ func addSchedulerName(pod *corev1.Pod, app *v1beta2.SparkApplication) *patchOper
 	return &patchOperation{Op: "add", Path: "/spec/schedulerName", Value: *schedulerName}
 }
 
+func addPriorityClassName(pod *corev1.Pod, app *v1beta2.SparkApplication) *patchOperation {
+	var priorityClassName *string
+
+	if app.Spec.BatchSchedulerOptions.PriorityClassName != nil {
```

Comment:
Looks like this if block is not needed and you can directly do:

```
priorityClassName := app.Spec.BatchSchedulerOptions.PriorityClassName
```

since `priorityClassName` is checked below.

Reply:
My bad! We must test the existence of block `batchSchedulerOptions` before getting the `priorityClassName`.